src/common/contentPreference.ts

```diff
@@ -278,3 +373,37 @@ export const unfollowEntity = ({
       throw new Error('Entity not supported');
   }
 };
+
+export const blockEntity = async ({
+  ctx,
+  id,
+  entity,
+}: {
+  ctx: AuthContext;
+  id: string;
+  entity: ContentPreferenceType;
+}): Promise<void> => {
+  switch (entity) {
+    case ContentPreferenceType.User:
+      return blockUser({ ctx, id });
+    case ContentPreferenceType.Keyword:
+      return blockKeyword({ ctx, id });
+    case ContentPreferenceType.Source:
+      return blockSource({ ctx, id });
+    default:
+      throw new Error('Entity not supported');
+  }
+};
+
+export const unblockEntity = async ({
+  ctx,
+  id,
+  entity,
+}: {
+  ctx: AuthContext;
+  id: string;
+  entity: ContentPreferenceType;
+}): Promise<void> => {
+  // currently unblock is just like unfollow, eg. remove everything from db
```

Comment:
Why remark, is it going to change?

Reply:
Not sure currently, but thought note is due here since it might look as an error to use `unfollow`.